src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/MsBuild/MsBuildUtilitiesTest.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    [ProjectSystemTrait]
+    public class MsBuildUtilitiesTest
+    {
+        [Fact]
+        public void MsBuildUtilities_GetProperty_MissingProperty()
+        {
+            string projectXml =
+@"<Project>
+  <PropertyGroup>
+     <MyProperty>MyPropertyValue</MyProperty>
+  </PropertyGroup>
+</Project>";
+
+            using (var project = new MsBuildTempProjectFile(projectXml))
+            {
+                var property = MsBuildUtilities.GetProperty(project.Project, "NonExistantProperty");
+                Assert.Null(property);
+            }
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetProperty_ExistentProperty()
+        {
+            string projectXml =
+@"<Project>
+  <PropertyGroup>
+     <MyProperty>MyPropertyValue</MyProperty>
+  </PropertyGroup>
+</Project>";
+
+            using (var project = new MsBuildTempProjectFile(projectXml))
+            {
+                var property = MsBuildUtilities.GetProperty(project.Project, "MyProperty");
+                Assert.NotNull(property);
+            }
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_SingleValue()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("MyPropertyValue");
+            Assert.Equal(1, values.Length);
+            Assert.Equal("MyPropertyValue", values[0]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_MultipleValues()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1;2");
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_NonDefaultDelimiter()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1|2", '|');
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_EmptyValues()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1;   ;;;2");
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetOrAddProperty_NoGroups()
+        {
+            using (var project = new MsBuildTempProjectFile())
+            {
+                MsBuildUtilities.GetOrAddProperty(project.Project, "MyProperty");
```

Comment:
That one was intentional. I wanted to validate the project itself rather than the individual element to circle back and make sure that the element did really get added.